pkg/reconciler/common/proxy.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package common
+
+import (
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func ApplyProxySettings(u *unstructured.Unstructured) error {
+	if u.GetKind() != "Deployment" {
+		// Don't do anything on something else than Deployment
+		return nil
+	}
+
+	var proxyEnv = []corev1.EnvVar{{
+		Name:  "HTTPS_PROXY",
+		Value: os.Getenv("HTTPS_PROXY"),
+	}, {
+		Name:  "HTTP_PROXY",
+		Value: os.Getenv("HTTP_PROXY"),
+	}, {
+		Name:  "NO_PROXY",
+		Value: os.Getenv("NO_PROXY"),
+	}}
+
+	m := u.Object
+	// spec.Template.spec.containers
+	containers, found, err := unstructured.NestedSlice(m, "spec", "template", "spec", "containers")
+	if err != nil {
+		return err
+	}
+	if !found {
+		// No containers in the deployment, it is weird but let's not fail
+		return nil
+	}
+	for _, c := range containers {
+		currentEnv, found, err := unstructured.NestedSlice(c.(map[string]interface{}), "env")
+		if err != nil {
+			return err
+		}
+		if !found {
+			currentEnv = []interface{}{}
+		}
+		for _, e := range proxyEnv {
+			if e.Value == "" {
+				continue
+			}
+			envvar := map[string]interface{}{
+				"name":  e.Name,
+				"value": e.Value,
+			}
+			currentEnv = append(currentEnv, envvar)
```

Comment:
Considering the scenario if there is an env already with the same name, there will two env then

Reply:
Indeed. There is also the case where the env are there and we want to remove them. Will update 😉